gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java

```diff
@@ -121,50 +125,133 @@ public static Builder builder(String name) {
     return new Builder(name);
   }
 
-  private Serie.Builder addGauge(String name, Gauge gauge, long timeStamp, Serie.Builder builder) {
-    return builder.values(timeStamp, name, gauge.getValue());
+  private void reportGauge(MetricContext context, String name, Gauge gauge, long timeStamp) {
+    this.influxDB.write(
+        this.database,
+        this.timeUnit,
+        getSerieBuilder(MetricRegistry.name(context.getName(), name)).values(timeStamp, gauge.getValue())
```

Comment:
This code is repeated many times. Creating a method pushValue(context, name, value, timestamp) would make things cleaner, what do you think?